TimeFormat.java

```diff
@@ -9,9 +14,21 @@ public static void main(String[] args) {
 		// It concatenates the empty string "" with the leftmost hour-digit. 
 		// It then concatenates the resulting string with the rightmost hour-digit,
 		// and then uses parseInt to cast the resulting string as an int.
-		int hours = Integer.parseInt("" + args[0].charAt(0) + args[0].charAt(1));
+		int hours;
 		// Does the same with the minutes part of the input.
-		int minutes = Integer.parseInt("" + args[0].charAt(3) + args[0].charAt(4));
+		String minutes;
         // Replace this comment with the rest of your code
+		String hourinput = args[0];
+		hours = Integer.parseInt(hourinput.substring(0,2));
+		minutes =hourinput.substring(3,5);
+		if(hours>12) {
+			System.out.println(hours-12 + ":" + minutes + " PM");
+		} else {
+			if (hours==12 && minutes!="00") {

```

Comment:
1. spacing
2. 12:00 is PM ..
3. You could have moved the inner if to be else if